specification/security/resource-manager/Microsoft.Security/preview/2015-06-01-preview/security.json

```diff
@@ -1770,6 +1887,130 @@
                 "sku"
             ]
         },
+        "TopologyList": {
+            "type": "object",
+            "properties": {
+                "value": {
+                    "type": "array",
+                    "readOnly": true,
+                    "items": {
+                        "$ref": "#/definitions/TopologyResource"
+                    }
+                },
+                "nextLink": {
+                    "readOnly": true,
+                    "type": "string",
+                    "description": "The URI to fetch the next page."
+                }
+            }
+        },
+        "TopologyResource": {
+            "type": "object",
+            "properties": {
+                "properties": {
+                    "x-ms-client-flatten": true,
+                    "readOnly": true,
+                    "$ref": "#/definitions/TopologyResourceProperties"
+                }
+            },
+            "allOf": [{
+                    "$ref": "#/definitions/Resource"
+                },
+                {
+                    "$ref": "#/definitions/Location"
+                }
+            ]
+        },
+        "TopologyResourceProperties": {
+            "type": "object",
+            "properties": {
+                "calculatedDateTime": {
+                    "type": "string",
+                    "format": "date-time",
+                    "readOnly": true,
+                    "description": "The UTC time on which the topology was calculated"
+                },
+                "resources": {
+                    "type": "array",
+                    "readOnly": true,
+                    "description": "Azure resources which are part of this topology resource",
+                    "items": {
+                        "$ref": "#/definitions/TopologySingleResource"
+                    }
+                }
+            }
+        },
+        "TopologySingleResource": {
+            "type": "object",
+            "properties": {
+                "resourceId": {
+                    "type": "string",
+                    "readOnly": true,
+                    "description": "Azure resource id"
+                },
+                "severity": {
+                    "type": "string",
+                    "readOnly": true,
+                    "description": "The security severity of the resource"
+                },
+                "recommendationsExist": {
+                    "type": "boolean",
+                    "readOnly": true,
+                    "description": "Indicates if the resource has security recommendations"
+                },
+                "internetFacing": {
+                    "type": "boolean",
+                    "readOnly": true,
+                    "description": "Indicates if the resource has direct connectivity to the Internet"
+                },
+                "topologyScore": {

```

Comment:
How does a customer interpret an integer value for the topologyScore? Is 5 good or does it indicate a problem? what is the range of supported values. String enum better for this? If you need integer, may be add description. can it be an enum?

Reply:
The goal of the score is for clustering and filtering in UI. Since we have customers with many resources, by default we show resources with higher score (lower score items will be less visible). also, resources with same score may be clustered into same UI object.
This is private preview concept and we may improve the explanation upon feedback from customers (before going public)